engine/src/flutter/impeller/display_list/aiks_dl_shadow_unittests.cc

```diff
@@ -0,0 +1,869 @@
+                                const DlPath& path,
+                                Scalar elevation,
+                                Scalar dpr,
+                                const DlPath* simple_path = nullptr) {
+  DlPath complex_path = MakeComplexPath(path);
+
+  builder.Save();
+
+  if (simple_path) {
+    builder.DrawShadow(*simple_path, DlColor::kBlue(), elevation, true, dpr);
+  }
+
+  builder.Translate(300, 0);
+  builder.DrawShadow(path, DlColor::kBlue(), elevation, true, dpr);
+
+  builder.Translate(300, 0);
+  builder.DrawShadow(complex_path, DlColor::kBlue(), elevation, true, dpr);
+
+  builder.Restore();
+  builder.Translate(0, 300);
+  builder.Save();
+
+  // Draw the mesh wireframe underneath the regular path output in the
+  // row above us.
+  builder.Translate(300, 0);
+  builder.DrawShadow(path, DlColor::kBlue(), elevation, true, dpr);
+  DrawShadowMesh(builder, path, elevation, dpr);
+
+  builder.Restore();
+}
+
+// Makes a Round Rect path using conics, but the weights on the corners is
+// off by just a tiny amount so the path will not be recognized.
+DlPath MakeAlmostRoundRectPath(const Rect& bounds,
+                               const RoundingRadii& radii,
+                               bool clockwise = true) {
+  DlScalar left = bounds.GetLeft();
+  DlScalar top = bounds.GetTop();
+  DlScalar right = bounds.GetRight();
+  DlScalar bottom = bounds.GetBottom();
+
+  // A weight of sqrt(2)/2 is how you really perform conic circular sections,
+  // but by tweaking it slightly the path will not be recognized as an oval
+  // and accelerated.
+  constexpr Scalar kWeight = kSqrt2Over2 - 0.0005f;
+
+  DlPathBuilder path_builder;
+  path_builder.MoveTo(DlPoint(right - radii.top_right.width, top));
+  path_builder.ConicCurveTo(DlPoint(right, top),
+                            DlPoint(right, top + radii.top_right.height),
+                            kWeight);
+  path_builder.LineTo(DlPoint(right, bottom - radii.bottom_right.height));
+  path_builder.ConicCurveTo(DlPoint(right, bottom),
+                            DlPoint(right - radii.bottom_right.width, bottom),
+                            kWeight);
+  path_builder.LineTo(DlPoint(left + radii.bottom_left.width, bottom));
+  path_builder.ConicCurveTo(DlPoint(left, bottom),
+                            DlPoint(left, bottom - radii.bottom_left.height),
+                            kWeight);
+  path_builder.LineTo(DlPoint(left, top + radii.top_left.height));
+  path_builder.ConicCurveTo(DlPoint(left, top),
+                            DlPoint(left + radii.top_left.width, top),  //
+                            kWeight);
+  path_builder.Close();
+  DlPath path = path_builder.TakePath();
+  if (!clockwise) {
+    path = ReflectPath(path);
+  }
+  return path;
+}
+}  // namespace
+
+TEST_P(AiksTest, DrawShadowDoesNotOptimizeHourglass) {
+  DisplayListBuilder builder;
+  builder.Clear(DlColor::kWhite());
+  builder.Scale(GetContentScale().x, GetContentScale().y);
```

Comment:
I thought we got rid of the need to do this, did you find you had to do it?

Reply:
Ugh, wonder what happened there, dreamtime coding session where I fixed all my annoyances?